### ISSUE TITLE
Discuss unique filenames for reports

### ISSUE THREAD

**Issue by BioPhoton** [code-pushup/cli#214]
This issue is here to collect information about the file name of the report and how we should think about it.

Michael:
The default name of a report can't be report.json as is can cause overwrites when running multiple times in a row (e.g. runtime assertions).
for multiple runs we should group the runs somehow
the report name can be custom
the upload command could upload multiple reports

Matej:
I think that report.json is a good and simple default for our end users. Being able to change the filename is something that's useful mainly for our tests - but only those that interact with the real file-system, tests using memfs can stick to the default, I believe.

Timestamp would be more useful than hash, I think, but I'm not sure it's a good idea at this point. And the idea with allowing placeholder syntax in the persist.filename option would then make sense, but there are unintended consequences I would prefer to avoid. E.g. the upload command would then have to resolve the filename dynamically and can have more than one match. But uploading multiple reports makes no sense at present and introduces unexpected behavior:

the upload command takes the current commit as the ID it send to the portal, so multiple reports would get matched to same commit ID and thus the "multi-upload" will result in only one report (the last one if uploaded in sequence, who knows which one if uploaded in parallel 😬)
we don't prompt the user to clean their persist.outputDir, which could have intended consequences of re-uploaded the same report when running collect and upload multiple times
Also, I think anything other than changing adding --persist.filename and using it to fix test flakiness should be out of scope of this PR. Changing the default and changing the single upload to a multi-upload ... it changes way too much and I'm not sure it's for the better.

**Code:** 

```typescript

/**
 * Regular expression for the default report filename.
 */
export const REPORT_NAME_PATTERN =
  /^(report\.)(\d{4}-\d{2}-\d{2}T\d{2}-\d{2}-\d{2}\.\d{3}Z)/;
export function reportFileName<T extends { date: string }>(report: T): string {
  return `report.${report.date.replace(/:/g, '-')}`;
}


export async function loadReports(options: PersistConfig) {
  const { outputDir, filename, format } = options;
  await ensureDirectoryExists(outputDir);

  const dirResult = await readdir(outputDir);

  return await Promise.allSettled(
    dirResult
      // filter by file extension
      .filter(file =>
        format ? format.find(ext => file.endsWith(`.${ext}`)) : true,
      )
      // filter by file name
      .filter(file =>
        filename
          ? file.includes(filename)
          : new RegExp(REPORT_NAME_PATTERN).test(file),
      )
      .map(file => getFileResult(file, outputDir)),
  );
}
```

Potential enhancement for another consideration when we more on that:

From LHCI:
```json
{
    reportFilenamePattern: {
      type: 'string',
      description: '[filesystem only] The pattern to use for naming Lighthouse reports.',
      default: '%%HOSTNAME%%-%%PATHNAME%%-%%DATETIME%%.report.%%EXTENSION%%',
    }
}
```
